examples/defaulter-gen/_output_tests/marker/zz_generated.go

```diff
@@ -0,0 +1,95 @@
+// +build !ignore_autogenerated
+
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by defaulter-gen. DO NOT EDIT.
+
+package marker
+
+import (
+	"encoding/json"
+	"reflect"
+
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// RegisterDefaults adds defaulters functions to the given scheme.
+// Public to allow building arbitrary schemes.
+// All generated defaulters are covering - they call all nested defaulters.
+func RegisterDefaults(scheme *runtime.Scheme) error {
+	scheme.AddTypeDefaultingFunc(&Defaulted{}, func(obj interface{}) { SetObjectDefaults_Defaulted(obj.(*Defaulted)) })
+	scheme.AddTypeDefaultingFunc(&DefaultedWithFunction{}, func(obj interface{}) { SetObjectDefaults_DefaultedWithFunction(obj.(*DefaultedWithFunction)) })
+	return nil
+}
+
+func SetObjectDefaults_Defaulted(in *Defaulted) {
+	if reflect.ValueOf(in.Field).IsZero() {
+		in.Field = "bar"
+	}
+	if reflect.ValueOf(in.OtherField).IsZero() {
+		in.OtherField = 0
+	}
+	if reflect.ValueOf(in.List).IsZero() {
+		if err := json.Unmarshal([]byte(`["foo", "bar"]`), &in.List); err != nil {
```

Comment:
Hm, we can probably leave this for the first take, but I think we'll eventually want to pre-unmarshal and make shallow or deep copies for these lines as needed.

Reply:
I'll add this as an action item for future improvements.

Reply:
Can you open an issue and list the things that we need to improve?

Reply:
+1, we've discussed pre-unmarshalling a couple times. It will definitely be on the follow up list.